homeassistant/components/inels/__init__.py

```diff
@@ -0,0 +1,134 @@
+"""The iNELS integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from inelsmqtt import InelsMqtt
+from inelsmqtt.discovery import InelsDiscovery
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from .const import BROKER, BROKER_CONFIG, DEVICES, DOMAIN, LOGGER, OLD_ENTITIES
+
+PLATFORMS: list[Platform] = [
+    Platform.COVER,
+]
+
+
+async def _async_config_entry_updated(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Call when config entry being updated."""
+
+    client: InelsMqtt = hass.data[BROKER]
+
+    await hass.async_add_executor_job(client.disconnect)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up iNELS from a config entry."""
+
+    if CONF_HOST not in entry.data:
+        LOGGER.error("MQTT broker is not configured")
+        return False
+
+    inels_data: dict[str, Any] = {
+        BROKER_CONFIG: entry.data,
+    }
+
+    if hass.data.get(DOMAIN) is None:
+        hass.data.setdefault(DOMAIN, {})[entry.entry_id] = inels_data
```

Comment:
You shouldn't be adding to hass.data until after you verify you can connect and authenticate.

Reply:
Where you have it below is fine on line 88

Reply:
Is my fix correct?